src/main/java/teammates/common/util/RecaptchaVerifier.java

```diff
@@ -0,0 +1,86 @@
+package teammates.common.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.util.stream.Collectors;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+/**
+ * Used to handle the verification of the user's reCAPTCHA response.
+ *
+ * @see <a href="https://developers.google.com/recaptcha/docs/verify">reCAPTCHA user response verification API</a>
+ * @see <a href="https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-tests-with-recaptcha-what-should-i-do">Automated testing with reCAPTCHA v2</a>
+ */
+public final class RecaptchaVerifier {
+
+    /** The Google reCAPTCHA API URL to verify the response token. */
+    public static final String VERIFY_URL = "https://www.google.com/recaptcha/api/siteverify";
+
+    /** The shared secret key between the TEAMMATES site and reCAPTCHA. */
+    public static final String SECRET_KEY = Config.CAPTCHA_SECRET_KEY;
+
+    private static final Logger log = Logger.getLogger();
+
+    private RecaptchaVerifier() {
```

Comment:
> Note: need to call `RecaptchaVerifier` in `LinkRecoveryAction` (in #9413). This can either be done in this PR or the other one (depending on which one is merged first)

The private constructor is to prevent instantiation